FLT/Mathlib/Topology/Algebra/Module/Equiv.lean

```diff
@@ -50,3 +54,78 @@ def ContinuousLinearEquiv.piScalarPiComm {α β : Type*} (R : α → Type*) (φ
     [(a : α) → (b : β) → Module (R a) (φ a b)] [(a : α) → (b : β) → TopologicalSpace (φ a b)] :
     ((a : α) → (b : β) → φ a b) ≃L[∀ a, R a] ((b : β) → (a : α) → φ a b) where
   __ := LinearEquiv.piScalarPiComm R φ
+
+lemma ContinuousLinearEquiv.toContinuousAddEquiv_trans
+    {R : Type*} {E : Type*} [Semiring R] [AddCommMonoid E] [Module R E]
+    [TopologicalSpace E] {e f : E ≃L[R] E} :
+    (e.trans f).toContinuousAddEquiv =
+      e.toContinuousAddEquiv.trans f.toContinuousAddEquiv := rfl
+
+lemma ContinuousLinearEquiv.toMatrix_det_ne_zero
+ {F : Type*} [Field F] {ι : Type*} [Fintype ι] [DecidableEq ι]
+  {V : Type*} [AddCommGroup V] [Module F V] [TopologicalSpace V]
+    (b : Module.Basis ι F V) (ρ : V ≃L[F] V) :
+      (ρ.toLinearMap.toMatrix b b).det ≠ 0 := by
+      simp[LinearEquiv.det_ne_zero]
```

Comment:
This proof also works fine with `[CommRing F] [Nontrivial F]` and everything else still compiles. All this does make me wonder whether you're focussing too much on x != 0 when you can get away with IsUnit x but I've not read the proof yet so maybe I'm wrong.